README.md

```diff
@@ -1,2 +1,104 @@
+2. [Features](#features)
+    1. [Publisher](#publisher)
+    2. [Retriever](#retriever)
+    3. [Key Rotation](#key-rotation)
+3. [Where to get it](#where-to-get-it)
+4. [How to use](#how-to-use)
+    1. [Publisher](#publisher-1)
+    2. [Retriever](#retriever-1)
+5. [Dependencies](#dependencies)
+6. [Contributing](#contributing-to-key-switcheroo)
+
+
+## What is it?
+
+**key-switcheroo** is a Python package that provides tools for **easy** :smile:, **reliable** :white_check_mark:, and **secure** :lock: SSH key management. The package contains tools to allow users to generate SSH public/private key pairs and securely store the public key either on AWS S3 or locally based on user preferences. The package also contains a tool used by the server host to retrieve and match the public keys with the corresponding private key during SSH connection attempts. Additionally, the package supports a feature for periodic rotation and swapping of public keys to enhance security.
+
+
+## Features
+
+### Publisher
+
+The *publisher* tool offers a user-friendly interface to securely generate SSH public/private key pairs. It allows users to specify the storage location for the public key, either on AWS S3 or on their local machine. The generated private key is stored securely and can be used for SSH authentication.
+
+### Retriever
+
+The *retriever* tool is designed to be used by server hosts for retrieving the public keys stored by the *publisher*. When an SSH connection attempt is made, the *retriever* fetches the corresponding public key associated with the private key used in the connection attempt. The tool compares the retrieved public key with the provided public key, ensuring a secure and authenticated connection.
+
+### Key rotation
+
+To enhance security, **key-switcheroo** supports a key rotation feature. The user simply needs to call the publisher script again with the same credentials and the program will swap and rotate the stored public keys. This process helps mitigate the risks associated with long-term key exposure and strengthens the overall security posture.
+
+
+## Where to get it
+
+The source code is currently hosted on GitHub at: https://github.com/SSH-key-rotation-AWS/team-henrique
+
+Binary installer for the latest released version is available at the [Python Package Index (PyPI)](https://pypi.org/project/key-switcheroo/).
+
+`pip install key-switcheroo`
+
+
+## How to use
+
+Once the package is installed, commands can be called from the user's CLI for both the *publisher* and *retriever* using different optional arguments.
+
+For help with command-line arguments,
+
+`switcheroo --help` or `switcheroo -h`
+
+### Publisher
+
+When using the *publisher* for creating and publishing new SSH keys, the user has a couple of different *optional* arguments, in addition to the *required* arguments.
+
+**Required Arguments:**
+1. `hostname` - host server
+2. `user` - username of the connecting client
+
+**Optional Arguments:**
```

Comment:
how are we handling the AWS authentication credentials? (don't we need that to be configured as well?)

Reply:
The plan is to leave it up to the user executing the script to have AWS CLI configured, and if that is configured correctly all is fine